src/components/Trade/hooks/useSwapper/useSwapper.ts

```diff
@@ -117,13 +121,22 @@ export const useSwapper = () => {
 
     if (!minimumQuote) return
 
+    // to account for gas fees when trading Max ETH
+    const gasDeduction =
```

Comment:
We want to always account for gas fees, not only when sending max.
See the review comment.

Reply:
Right, sorry, not sure how I missed that. Will fix asap

Reply:
Hi @gomesalexandre, I think I have resolved the issue now, however I believe the `estimatedGasFees` value in the `TradeInput` component is higher than the actual estimated gas fee used to create the quote for the transaction so the Max button doesn't result in a very good actual Max and the check to see if there is enough ETH for fees is using the significantly higher gas estimate. I've attached a screenrecording with the `estimatedGasFees` var being logged in the console

https://user-images.githubusercontent.com/14836056/164868363-b437fa7b-872e-4841-9c55-18cd22336108.mov

Reply:
> Hi @gomesalexandre, I think I have resolved the issue now, however I believe the `estimatedGasFees` value in the `TradeInput` component is higher than the actual estimated gas fee used to create the quote for the transaction so the Max button doesn't result in a very good actual Max and the check to see if there is enough ETH for fees is using the significantly higher gas estimate. I've attached a screenrecording with the `estimatedGasFees` var being logged in the console
> Screen.Recording.2022-04-23.at.5.31.39.AM.mov

@brymut tried your implementation and  I'm consistently getting "Not enough ETH to cover gas" error on the first try. It only works when clicking "Max" a second time.

Reply:
I think I have addressed the issue @gomesalexandre, problem was that estimatedGasFees was still undefined and wasn't being factored in, I have added that uses the quote fees to address this. 

https://user-images.githubusercontent.com/14836056/165241979-921f9372-ee7d-49e1-93aa-d3f184f7d299.mov